src/main/kotlin/cacophony/codegen/linearization/BasicBlock.kt

```diff
@@ -7,5 +7,5 @@ class BasicBlock(
     val label: BlockLabel,
     val instructions: List<Instruction>,
     val successors: Set<BasicBlock>,
-    val predecessors: Set<BasicBlock>,
+    val predecessors: MutableSet<BasicBlock>,
```

Comment:
otherwise it's impossible to properly instantiate `BasicBlock` because of cyclic dependencies (to create predecessor I need its successor, for which I need successor's predecessor)